### PR TITLE
bpo-42316: Allow unparenthesized walrus operator in indexes

### PR DESCRIPTION
**Pull request by lysnikolaou**
https://bugs.python.org/issue42316

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2020-11-16-17-30-03.bpo-42316._DdmpQ.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Allow an unparenthesized walrus in subscript indexes.
```
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2020-11-16-17-30-03.bpo-42316._DdmpQ.rst:1]: Is this a regression? In that case the news entry should mention it (so is clear that is not new behaviour).

**Review by pablogsal — APPROVED** [review]

File: Misc/NEWS.d/next/Core and Builtins/2020-11-16-17-30-03.bpo-42316._DdmpQ.rst (hunk 1 above)
Comment (lysnikolaou) [reply Misc/NEWS.d/next/Core and Builtins/2020-11-16-17-30-03.bpo-42316._DdmpQ.rst:1]: Nope, this is no regression. It's new syntax. It was proposed by @gvanrossum [here](https://github.com/python/cpython/pull/23291#discussion_r523908212).

Comment (pablogsal) [reply Misc/NEWS.d/next/Core and Builtins/2020-11-16-17-30-03.bpo-42316._DdmpQ.rst:1]: 👍

File: Lib/test/test_named_expressions.py [hunk 2]
```diff
@@ -271,6 +271,12 @@ def test_named_expression_assignment_16(self):
         fib = {(c := a): (a := b) + (b := a + c) - b for __ in range(6)}
         self.assertEqual(fib, {1: 2, 2: 3, 3: 5, 5: 8, 8: 13, 13: 21})
 
+    def test_named_expression_assignment_17(self):
+        a = [1]
+        element = a[b:=0]
+        self.assertEqual(b, 0)
+        self.assertEqual(element, a[0])
+
```
Comment (pablogsal) [inline Lib/test/test_named_expressions.py:279]: Can you add a test also with more complex indexing like [b:=0, c:=1]?

Comment (lysnikolaou) [reply Lib/test/test_named_expressions.py:279]: We can only test that this does not raise a `SyntaxError`, right? Is there a builtin data structure where we can use indexing with a tuple?

Comment (pablogsal) [reply Lib/test/test_named_expressions.py:279]: > We can only test that this does not raise a `SyntaxError`, right?

Yup. You can either use a class with custom `__getitem__` or a mock

Comment (lysnikolaou) [reply Lib/test/test_named_expressions.py:279]: Done.

**Review by pablogsal — APPROVED** [review]

**Pull request merged**